src/lib.rs

```diff
@@ -168,6 +168,56 @@ impl Order {
         self.account.get(&mut self.nonce, challenge_url).await
     }
 
+    /// Deactivate the order's pending or valid authorizations
+    ///
+    /// Returns the updated status of the order post-deactivation.
+    ///
+    /// This is useful when you want to cancel a pending order you wish to abandon, or to revoke
+    /// valid authorizations for the order's identifiers to prevent reuse.
+    ///
+    /// Once deactivated the order and associated resources (authorizations, challenges, etc) can
+    /// not be updated further.
+    pub async fn deactivate(&mut self) -> Result<&OrderState, Error> {
```

Comment:
This feels like a comparatively fat abstraction. How do you feel about attaching something like this to the `Authorization` type directly? `Order` already has an `authorizations()` method... You'd have to `refresh()` by hand.

Reply:
How about doing something like this:

```rust
struct Order {
     authorizations: Vec<Authorization>,
}

struct Authorization {
     pub url: String,
     pub state: Option<AuthorizationState>,
     // maybe:
     account: Arc<AccountInner>,
}

impl Authorization {
     pub async fn refresh(&mut self) -> Result<&AuthorizaionState, Error> { .. }

     pub async fn deactivate(&mut self) -> Result<(), Error> { .. }
}
```

What do you think?

Reply:
That seems like a nice solution. I'll give it a go :+1:

Reply:
Here's a first pass: 319be2335821718e38fd51ac125de8106e91ee76

I more or less matched the approach `Order` and `OrderState` use (e.g. accessors vs `pub` fields). It feels to me like the existing behaviour of `Order::authorizations()` where you got back the state immediately for each authz made sense, so I implemented `Authorization` to hold non-optional `AuthorizationState` instead of `Option<AuthorizationState`. If you think it's better to populate lazily and have `authorizations()` return `state: None` I can rework.

With the nonce stuff factored in it didn't seem like `RefCell` added much, the lifetimes in use seemed OK, but I'm open to further iteration if you have suggestions!